diff --git a/Makefile b/Makefile
index 555d1c46..a725af82 100644
--- a/Makefile
+++ b/Makefile
@@ -37,7 +37,10 @@ LAPACK_NOOPT := $(filter-out -O0 -O1 -O2 -O3 -Ofast,$(LAPACK_FFLAGS))
 SUBDIRS_ALL = $(SUBDIRS) test ctest utest exports benchmark ../laswp ../bench cpp_thread_test
 
 .PHONY : all libs netlib $(RELA) test ctest shared install
-.NOTPARALLEL : all libs $(RELA) prof lapack-test install blas-test
+.NOTPARALLEL : all libs netlib $(RELA) tests shared prof lapack-test install blas-test
+
+shared: libs netlib $(RELA)
+tests: libs netlib $(RELA) shared
 
 all :: libs netlib $(RELA) tests shared
 	@echo
